Modules/Tracker/LinePool.lua

```diff
@@ -68,35 +79,64 @@ function LinePool.Initialize(trackedQuestsFrame, UntrackQuest, TrackerUpdate)
         line.SetMode = _SetMode
 
         function line:SetZone(ZoneId)
-            self.ZoneId = TrackerUtils:GetZoneNameByID(ZoneId)
-            self.expandZone.zoneId = ZoneId
+            if ZoneId == "Achievements" then
+                self.expandZone.zoneId = ZoneId
+            else
+                self.ZoneId = TrackerUtils:GetZoneNameByID(ZoneId)
+                self.expandZone.zoneId = ZoneId
+            end
         end
 
         function line:SetQuest(Quest)
+            if type(Quest) ~= "table" then
+                local questID = Quest
+                Quest = {
+                    Id = questID
+                }
+            end
             self.Quest = Quest
             self.expandQuest.questId = Quest.Id
+
+            -- Set Timed Quest Flag
+            if Quest.trackTimedQuest then
+                self.trackTimedQuest = Quest.trackTimedQuest
+            end
         end
 
         function line:SetObjective(Objective)
             self.Objective = Objective
         end
 
+        line.OnUpdate = function(self, elapsed)
+            timeElapsed = timeElapsed + elapsed
+            if timeElapsed > 1 and self.trackTimedQuest then
+                _QuestieQuestTimers:UpdateTimerFrame()
```

Comment:
I think we should move the `UpdateTimerFrame` function to the main module if we need it outside instead of adding `.private` stuff outside of it.